### PR TITLE
Registration app

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: registration/models.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    hcw_msisdn = models.CharField(max_length=255, null=True, blank=True)
+    mom_msisdn = models.CharField(max_length=255, null=False, blank=False)
+    mom_id_type = models.CharField(max_length=8, null=False, blank=False)
+    mom_lang = models.CharField(max_length=3, null=False, blank=False)
+    mom_edd = models.DateField(null=True, blank=True)
+    mom_id_no = models.CharField(max_length=100, null=True, blank=True)
+    mom_dob = models.DateField(null=True, blank=True)
+    clinic_code = models.CharField(max_length=100, null=True, blank=True)
+    authority = models.CharField(max_length=6, null=False, blank=False)
```
Comment (gsvr) [inline registration/models.py]: @imsickofmaps Please suggest a better name here?

File: control/settings.py [hunk 2]
```diff
@@ -122,6 +122,7 @@ def abspath(*args):
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
+    # 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
```
Comment (imsickofmaps) [inline control/settings.py]: remove

File: control/settings.py [hunk 3]
```diff
@@ -317,6 +321,20 @@ def abspath(*args):
     # 'dsn': 'http://public:secret@example.com/1',
 }
 
+# REST Framework conf defaults
+REST_FRAMEWORK = {
+    'DEFAULT_PERMISSION_CLASSES': ('rest_framework.permissions.IsAdminUser',),
+    'PAGINATE_BY': None,
```
Comment (imsickofmaps) [inline control/settings.py]: Let's make this `1000`

File: control/settings.py [hunk 4]
```diff
@@ -158,9 +159,12 @@ def abspath(*args):
     'djcelery_email',
     'tastypie',
     'bootstrapform',
+    'rest_framework',
+    'rest_framework.authtoken',
     # Custom apps
```
Comment (imsickofmaps) [inline control/settings.py:164]: Missing `'django_filters',`

File: registration/models.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+from django.db import models
+
+
+class Source(models.Model):
+    """ The source from which a registation originates.
+    """
+    name = models.CharField(max_length=100, null=False, blank=False)
+    created_at = models.DateTimeField(auto_now_add=True)
```
Comment (imsickofmaps) [inline registration/models.py:10]: Missing the FK to the user model so we can determine source from API key on POST request

File: registration/models.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+class Registration(models.Model):
+    """ A registation submitted via Vumi or external sources.
+        Upon saving a registration, 3 things should happen:
+        1. Send registration information to Jembi
+        2. Create or update the contact on Vumi
+        3. Create a subscription for the mom
+    """
+    hcw_msisdn = models.CharField(max_length=255, null=True, blank=True)
+    mom_msisdn = models.CharField(max_length=255, null=False, blank=False)
+    mom_id_type = models.CharField(max_length=8, null=False, blank=False)
```
Comment (imsickofmaps) [inline registration/models.py]: add `choices` contraint here

File: registration/models.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    mom_lang = models.CharField(max_length=3, null=False, blank=False)
```
Comment (imsickofmaps) [inline registration/models.py]: add `choices` contraint here

File: registration/models.py (hunk 1 above)
Comment (imsickofmaps) [inline registration/models.py]: This is fine. Just include a better doc string to explain.

File: registration/serializers.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from rest_framework import serializers
+
+from .models import Source, Registration
+
+
+class SourceSerializer(serializers.HyperlinkedModelSerializer):
+
+    class Meta:
+        model = Source
+        fields = ('url', 'id', 'name')
```
Comment (imsickofmaps) [inline registration/serializers.py]: update post model change

File: registration/tests.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+class TestRegistrationsAPI(AuthenticatedAPITestCase):
+
+    def test_create_source(self):
+        response = self.make_source()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        d = Source.objects.last()
+        self.assertEqual(d.name, 'Test Source')
+        self.assertEqual(d.id, 2)
```
Comment (imsickofmaps) [inline registration/tests.py]: You're not in control of this field. Don't test it.

File: registration/tests.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+    def test_create_registration(self):
+        reg_response = self.make_registration()
+        self.assertEqual(reg_response.status_code, status.HTTP_201_CREATED)
+
+        d = Registration.objects.last()
+        self.assertEqual(d.id, 1)
```
Comment (imsickofmaps) [inline registration/tests.py]: Missing a test for a field you're in control of

**Comment by gsvr** [comment]
@imsickofmaps Ready for re-review.

**Comment by imsickofmaps** [comment]
Just the minor docstring update then :+1:

**Pull request closed without merging**